test/integration/document_properties_spec.mjs

```diff
@@ -0,0 +1,110 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { closePages, loadAndWait } from "./test_utils.mjs";
+
+const FIELDS = [
+  "fileName",
+  "fileSize",
+  "title",
+  "author",
+  "subject",
+  "keywords",
+  "creationDate",
+  "modificationDate",
+  "creator",
+  "producer",
+  "version",
+  "pageCount",
+  "pageSize",
+  "linearized",
+];
+
+// Unicode bidi isolation characters, Fluent adds these markers to the text.
+const FSI = "\u2068";
+const PDI = "\u2069";
+
+describe("PDFDocumentProperties", () => {
+  function waitForDocumentProperties(page) {
+    return page.evaluate(() => [
+      new Promise(resolve => {
+        const { eventBus } = window.PDFViewerApplication;
+        eventBus.on("documentpropertiesrendered", () => resolve, {
+          once: true,
+        });
+      }),
+    ]);
+  }
+
+  async function getFieldProperties(page) {
+    const promises = [];
+
+    for (const name of FIELDS) {
+      promises.push(
+        page.evaluate(
+          n => [n, document.getElementById(`${n}Field`).textContent],
+          name
+        )
+      );
+    }
+    return Object.fromEntries(await Promise.all(promises));
+  }
+
+  describe("Document with both /Info and /Metadata", () => {
+    let pages;
+
+    beforeEach(async () => {
+      pages = await loadAndWait("basicapi.pdf", ".textLayer .endOfContent");
+    });
+
+    afterEach(async () => {
+      await closePages(pages);
+    });
+
+    it("must check that the document properties dialog has the correct information", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          await page.click("#secondaryToolbarToggleButton");
+          await page.waitForSelector("#secondaryToolbar", { hidden: false });
+
+          await page.click("#documentProperties");
+          await page.waitForSelector("#documentPropertiesDialog", {
+            hidden: false,
+          });
+
+          await waitForDocumentProperties(page);
```

Comment:
Can't we just await on having something in whatever property span ?

Reply:
Sure, that works too and it's less code.